[SAASINT-4016] DDS: iboss: (Crawler + Agent) Integration v1.0.0 (#20414)
DataDog/integrations-core: iboss/README.md

## Overview

> [iboss][1] provides secure internet access and Zero Trust enforcement for users anywhere. It combines key security functions such as Secure Web Gateway (SWG), CASB, ZTNA, and DLP into a single, scalable solution. Traffic is routed through iboss's infrastructure to ensure consistent policy enforcement and threat protection.

Reviewer: ```suggestion
[iboss][1] provides secure internet access and Zero Trust enforcement for users, wherever they are. It combines key security functions such as Secure Web Gateway (SWG), CASB, ZTNA, and DLP into a single, scalable solution. Traffic is routed through iboss's infrastructure to ensure consistent policy enforcement and threat protection.
```

Author: Done

> This integration parses and ingest the following types of logs:

Author: Done

> - **Web Logs** : Logs provide information about client requests to web resources, enabling monitoring of web traffic and policy enforcement.

Author: Done

> - **DLP Logs** : Logs provide information related to data loss prevention, tracking policy enforcement and potential sensitive data exposures.

Author: Done

> - **DLP Logs** : Provides information related to data loss prevention, tracking policy enforcement and potential sensitive data exposures.

Author: Done

> - **Audit Logs** : Provides information of user and system activities to provide traceability and support compliance monitoring.

Author: Done

> - **Audit Logs** : Logs provide information of user and system activities to provide traceability and support compliance monitoring.

Author: Done

> Visualize detailed insights into these logs through the out-of-the-box dashboards. Additionally, ready-to-use Cloud SIEM detection rules are available to help you monitor and respond to potential security threats effectively.

Author: Done

> This integration ingests the following metrics:

Author: Done

> - **Gateway Performance** : Provides insights into request volumes, resource utilization, processing times, and error counts to monitor the health and efficiency of iboss gateways.

Author: Done

> - **Gateway Performance** : Metrics provide insights into request volumes, resource utilization, processing times, and error counts to monitor the health and efficiency of iboss gateways.

Author: Done

> - **Bandwidth** : Provides information about data usage across domains, users, locations, and protocols, enabling monitoring of network traffic volume and flow patterns.

Author: Done

> - **Bandwidth** : Metrics provide information about data usage across domains, users, locations, and protocols, enabling monitoring of network traffic volume and flow patterns.

Author: Done

> - **Digital Experience** : Provides insights into user experience by measuring connection times between clients, proxies, and servers across users, assets, and locations.

Author: Done

> - **Web** : Provides visibility into user web activity, including site visits, blocks, malware detections, and search trends, to support usage analysis and policy effectiveness.

Author: Done

> - **Web** : Metrics provide visibility into user web activity, including site visits, blocks, malware detections, and search trends, to support usage analysis and policy effectiveness.

Author: Done

> - **CASB** : Provides insights into cloud app usage, user behavior, and traffic patterns, enabling visibility, threat detection, and enforcement of cloud access policies.

Author: Done

> - **Threat** : Metrics provide visibility into detected and prevented threats across users, assets, IPs, and geolocations, enabling threat trend analysis and risk monitoring.

Author: Done

> - **Zero Trust** : Metrics provide visibility into users, devices, resources, and traffic, enabling continuous monitoring, trust evaluation, and policy enforcement within the Zero Trust framework.

Author: Done

Visualize detailed insights into these metrics through the out-of-the-box dashboards. Additionally, monitors are provided to help get alerts on any potential issues.

## Setup

> **Note**: These steps are only required for collecting metrics.

Reviewer: ```suggestion
**Note**: The following steps are required only for collecting metrics. For log collection, see the [Log collection](#log-collection) section below.
```

Author: Done

### Generate API credentials in iboss

To collect metrics, you can either use an existing user with **Full Administrator** access to the **Reporting & Analytics** module, or create a custom user with a custom RBAC group by following the steps to set up reporting-only permissions and assign the user to that RBAC group.

#### Create New RBAC

1. Login to iboss portal as System Administrator.
2. Go to **Home** > **System Administrators**.
3. Switch to **Role-Based Access Control** tab.
4. Click on **Add Custom RBAC Group**.

> 5. Add **Display Name** for the RBAC.

Author: Done

6. In the **General Info & Permissions** tab, enable only the **Reporting & Analytics** option to limit permissions to reporting only. Next, go to the **Reporting & Analytics Permissions** tab and choose **Full Administrator** from the **Permission Type** dropdown to allow complete access within the reporting module.
7. Click on **Add RBAC Group**.

#### Create New User

> 1. Login to iboss portal as System Administrator.

Author: Done

2. Go to **Home** > **System Administrators**.
3. Click on **Add New System Administrator**.
4. Add details for **System Administrator Email Address**, **First Name**, and **Last Name**.
5. For **Use RBAC Groups**, select the RBAC group with minimal permissions.

> 6. Click on **Add New System Administrator**.

Author: Done

**Note**: Make sure that MFA is disabled for the user account used by this integration.

### Connect your iboss Account to Datadog

> 1. Add your iboss Email Address, and Password.

Author: Done

   | Parameters                        | Description                                                                          |
   | --------------------------------- | ------------------------------------------------------------------------------------ |
   | Email Address                     | The Email Address of your iboss account.                                             |
   | Password                          | The Password of your iboss account.                                                  |
   | Get Gateway Performance Metrics   | Enable to collect Gateway Performance Metrics from iboss. The default value is True. |
   | Get Bandwidth Metrics             | Enable to collect Bandwidth Metrics from iboss. The default value is True.           |
   | Get Digital Experience Metrics    | Enable to collect Digital Experience Metrics from iboss. The default value is True.  |
   | Get Web Metrics                   | Enable to collect Web Metrics from iboss. The default value is True.                 |
   | Get CASB Metrics                  | Enable to collect CASB Metrics from iboss. The default value is True.                |

>    | Get Threat Metrics                | Enable to collect Threat Metrics from iboss. The default value is True.              |

Author: Done

   | Get Zero Trust Metrics            | Enable to collect Zero Trust Metrics from iboss. The default value is True.          |

2. Click **Save**.

### Installation

> **Note**: This steps are only required for collecting logs.

Author: Done

> **Note**: These steps are only required for collecting logs.

Reviewer 2: can you ping agent integrations team to verify this part?

Author: Adding @torosmassa to help here.

Reviewer 3: @steveny91 can you please have a look at this Assets PR, overall, from an Agent Ints perspective since you reviewed the TDD. Thank you!

To install the iboss integration, run the following Agent installation command in your terminal, then complete the configuration steps below. For more information, see the [Integration Management][4] documentation.

> **Note**: This step is not necessary for Agent version >= 7.68.0 .

Author: Done

```shell
sudo -u dd-agent -- datadog-agent integration install datadog-iboss==1.0.0
```

### Configuration

#### Log collection

1. Collecting logs is disabled by default in the Datadog Agent. Enable it in `datadog.yaml`:

   ```yaml
   logs_enabled: true
   ```

> 2. Add this configuration block to your `iboss.d/conf.yaml` file to start collecting your logs.

Author: Done. I’ve updated the content as you mentioned, but haven’t included the `datadog-agent` path since it’s OS-dependent.

>    See the sample [iboss.d/conf.yaml][6] for available configuration options.

Author: Done

>    See the sample [iboss.d/conf.yaml][5] for available configuration options.

Reviewer: ```suggestion
   See the sample configuration file ([iboss.d/conf.yaml][5]) for available options.
```

Author: Done

   ```yaml
   logs:
     - type: tcp # or 'udp'
       port: <PORT>
       source: iboss
       service: iboss
   ```

   **Note**:

   - `PORT`: Port should be similar to the port provided in **Configure syslog message forwarding from iboss** section.

>    - It is recommended not to change the service and source values, as these parameters are integral to the pipeline's operation.

Author: Done

3. [Restart the Agent][2].

#### Configure syslog message forwarding from iboss

> 1. Login in to the iboss Portal.

Author: Done

2. Navigate to **Integration Marketplace**, then select **Log Forwarding** from the left-hand menu and click the **Configure** button associated with the **Syslog Log Forwarding** widget.
3. Click the **Add Integration** button to add the Syslog integration.
4. Configure the settings as follows:
   - **Froward From**: Select **Reporter** from the dropdown.
   - **Select Reporting Database**: Select the Primary Reporting Database.

>    - **Service Name**: Choose any friendly name for the integration.

Author: Done

   - **Enable Service**: Set this to Enabled.
   - **Log Type**: Select **URL** from the dropdown.
   - **Protocol Type**: Select **UDP** or **TCP** from the dropdown.
   - **Syslog Facility Level**: Select **Facility Syslog** from the dropdown.
   - **Reporting Group**: Select **All** from dropdown.
   - **Host Name**: Enter the fully qualified domain name or IP address of the syslog server.
   - **Port**: Enter the port.
   - **Log Format**: Select **JSON** from the dropdown.
   - **Transfer Interval**: Select **Continuous** from the dropdown.
   - **Field Delimiter**: Select **SPACE** from the dropdown.
   - **Send DLP/Web/DNS/Malware/Audit Logs**: Set to Enable based on the preference for sending logs.

>    - **Fields to Forward**: Add all the fields except **DLP Base64 Encoded Meta Data**, **Base64 Encoded Meta Data** and **Chat GPT Message**.

Author: Done

     <br>After entering the required details, click **Add Service**.

### Validation

[Run the Agent's status subcommand][6] and look for `iboss` under the Logs Agent section.

## Data Collected

### Logs

| Format                    | Event Types                                      |
| ------------------------- | ------------------------------------------------ |
| JSON                      | Web Logs, DLP Logs, Audit Logs                   |

### Metrics

The iboss integration collects and forwards Gateway Performance, Bandwidth, Digital Experience, Web, CASB, Threat, CASB, and Zero Trust metrics to Datadog.

{{< get-metrics-from-git "iboss" >}}

### Events

The iboss integration does not include any events.

## Troubleshooting

### Permission denied while port binding

If you see a **Permission denied** error while port binding in the Agent logs:

> 1. Binding to a port number under 1024 requires elevated permissions. Grant access to the port using the `setcap` command:

Reviewer: ```suggestion
1. Binding to a port number under 1024 requires elevated permissions. Grant the necessary permissions using the `setcap` command:
```

Author: Done

   ```shell
   sudo setcap CAP_NET_BIND_SERVICE=+ep /opt/datadog-agent/bin/agent/agent
   ```

2. Verify the setup is correct by running the `getcap` command:

   ```shell
   sudo getcap /opt/datadog-agent/bin/agent/agent
   ```

>    With the expected output:

Reviewer: ```suggestion
   You should see output similar to:
```
